### PR TITLE
Import additional fields for Debian packages and optimizations

### PR DESCRIPTION
**Pull request by mateiw**
## What does this PR change?

- Import additional fields for Debian packages, notably Multi-Arch.
- Performance optimization of the generation of `Packages` file for Debian repos.

## GUI diff

No difference.

## Documentation
- No documentation needed

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Fixes https://github.com/uyuni-project/uyuni/issues/1030
Upstream issue https://github.com/SUSE/spacewalk/issues/9221

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: schema/spacewalk/common/tables/rhnPackageExtraTag.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                    CONSTRAINT rhn_pkg_extratags_pid_fk
+                       REFERENCES rhnPackage (id)
+                       ON DELETE CASCADE,
+    key_id      NUMBER NOT NULL
+                       REFERENCES rhnPackageExtraTagKey (id)
+                       ON DELETE CASCADE,
+    value       VARCHAR2(2048) NOT NULL,
+    created     timestamp with local time zone
+                    DEFAULT (current_timestamp) NOT NULL,
+    primary key (package_id, name)
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/rhnPackageExtraTag.sql]: name does not exist anymore

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+CREATE TABLE IF NOT EXISTS rhnPackageExtraTagKey
+(
+    id          NUMBER NOT NULL
+                    CONSTRAINT rhn_pkg_extra_tags_keys_id_pk PRIMARY KEY,
+    name        VARCHAR2(256) NOT NULL,
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: I would suggest to add a unique index on name. I think we will search for it a lot.

File: schema/spacewalk/common/tables/rhnPackageExtraTag.sql (hunk 1 above)
Comment (mateiw) [reply schema/spacewalk/common/tables/rhnPackageExtraTag.sql]: Fixed in #1327.

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql (hunk 2 above)
Comment (mateiw) [reply schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: Added index in #1237.

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                    DEFAULT (current_timestamp) NOT NULL
+)
+;
+
+CREATE SEQUENCE rhn_package_extra_tags_keys_id_seq;
+
+CREATE INDEX rhn_pkg_extra_tag_key_idx
+    ON rhnPackageExtraTagKey (name)
+    TABLESPACE [[32m_tbs]]
+    NOLOGGING;
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: We are not using oracle anymore. I hope we could just remove these two lines.

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                    CONSTRAINT rhn_pkg_extra_tags_keys_id_pk PRIMARY KEY,
+    name        VARCHAR2(256) NOT NULL,
+    created     timestamp with local time zone
+                    DEFAULT (current_timestamp) NOT NULL
+)
+;
+
+CREATE SEQUENCE rhn_package_extra_tags_keys_id_seq;
+
+CREATE INDEX rhn_pkg_extra_tag_key_idx
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: No "unique" index?

File: schema/spacewalk/upgrade/susemanager-schema-4.1.0-to-susemanager-schema-4.1.1/001-rhnPackageExtraTag.sql.postgresql [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                       ON DELETE CASCADE,
+    key_id      NUMERIC NOT NULL
+                       REFERENCES rhnPackageExtraTagKey (id)
+                       ON DELETE CASCADE,
+    value       VARCHAR(2048) NOT NULL,
+    created     TIMESTAMPTZ
+                     DEFAULT (current_timestamp) NOT NULL,
+    primary key (package_id, key_id)
+)
+;
```
Comment (mcalmer) [inline schema/spacewalk/upgrade/susemanager-schema-4.1.0-to-susemanager-schema-4.1.1/001-rhnPackageExtraTag.sql.postgresql]: the changes made above, need also go here. (missing the index)

Comment (mateiw) [reply schema/spacewalk/upgrade/susemanager-schema-4.1.0-to-susemanager-schema-4.1.1/001-rhnPackageExtraTag.sql.postgresql]: done

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql (hunk 4 above)
Comment (mateiw) [reply schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: done

File: schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql (hunk 3 above)
Comment (mateiw) [reply schema/spacewalk/common/tables/rhnPackageExtraTagKey.sql]: Removed lines.

**Comment by rpasche** [comment]
This sound very good. Can you tell or estimate, when Uyuni 4.1.1 will be released?

**Comment by cbbayburt** [comment]
> This sound very good. Can you tell or estimate, when Uyuni 4.1.1 will be released?

@rpasche the next release is by the end of this month, right before FOSDEM

**Comment by rpasche** [comment]
Great. Looking forward to upgrading then...

**Comment by rpasche** [comment]
@cbbayburt Sorry to bother again....just for clarification....will the "next" release (this month) be v4.1.1? (which seem to be the first version including *this* code)?

**Comment by cbbayburt** [comment]
> @cbbayburt Sorry to bother again....just for clarification....will the "next" release (this month) be v4.1.1? (which seem to be the first version including _this_ code)?

@rpasche the version will actually be 202001. The versioning is changed to YYYYMM format.

**Pull request merged**